### ISSUE TITLE
Blur event inside form can cause "no event handler associated with this event" exception

### ISSUE THREAD

**Issue by SteveSandersonMS** [dotnet/aspnetcore#21241]
Scenario:

```razor
<EditForm OnValidSubmit="NavigateToAnotherPage">
    <input @blur="@(() => {})" />
</EditForm>
```

Pressing "enter" inside the `input` causes:

```
System.ArgumentException: There is no event handler associated with this event. EventId: '45'.
Parameter name: eventHandlerId
  at Microsoft.AspNetCore.Components.RenderTree.Renderer.DispatchEventAsync (:44300/System.UInt64 eventHandlerId, Microsoft.AspNetCore.Components.RenderTree.EventFieldInfo fieldInfo, System.EventArgs eventArgs) <0x3124c68 + 0x00054> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer.DispatchEventAsync (:44300/System.UInt64 eventHandlerId, Microsoft.AspNetCore.Components.RenderTree.EventFieldInfo eventFieldInfo, System.EventArgs eventArgs) <0x31248a0 + 0x00078> in <filename unknown>:0 
  at :44300/Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer.ProcessNextDeferredEventAsync () <0x3477c48 + 0x000a6> in <filename unknown>:0 
    at Object.endInvokeDotNetFromJS (blazor.webassembly.js:1)
    at Object.invokeJSFromDotNet (blazor.webassembly.js:1)
```

Looks like the "navigate away" gets processed before the "blur", so the component has been disposed by the time we process its "blur" event handler.

**Comment by boindiil** [comment]
@SteveSandersonMS are there any plans to include this to release 3.2?

**Comment by SteveSandersonMS** [comment]
Sorry, the deadline for that has already passed. We're *really* close to the 3.2 release now!

**Comment by jameschch** [comment]
Can recreate similar error message in 3.2.0-rc1 with button click handler but only after publish to IIS host:

```
    <EditForm Model="@Config" OnValidSubmit="@ValidSubmit">
        <DataAnnotationsValidator />
        <ValidationSummary />

        <div class="form-group">
            <div class="row">
                <div class="col-sm-12 d-flex justify-content-center">
                    <button type="button" @onclick="@LoadSample" class="btn btn-secondary ml-2">Load Sample</button>
```

```
blazor.webassembly.js:1 Uncaught (in promise) Error: System.ArgumentException: There is no event handler associated with this event. EventId: '8'.
Parameter name: eventHandlerId
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x257c3a0 + 0x000ce> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---

  at Microsoft.JSInterop.Infrastructure.DotNetDispatcher.InvokeSynchronously (Microsoft.JSInterop.JSRuntime jsRuntime, Microsoft.JSInterop.Infrastructure.DotNetInvocationInfo& callInfo, Microsoft.JSInterop.Infrastructure.IDotNetObjectReference objectReference, System.String argsJson) <0x2f372b0 + 0x00180> in <filename unknown>:0 
  at Microsoft.JSInterop.Infrastructure.DotNetDispatcher.BeginInvokeDotNet (Microsoft.JSInterop.JSRuntime jsRuntime, Microsoft.JSInterop.Infrastructure.DotNetInvocationInfo invocationInfo, System.String argsJson) <0x2f36e78 + 0x00092> in <filename unknown>:0 
    at Object.endInvokeDotNetFromJS (https://optimizer.ml/_framework/blazor.webassembly.js:1:10117)
    at Object.invokeJSFromDotNet (https://optimizer.ml/_framework/blazor.webassembly.js:1:9737)
    at _mono_wasm_invoke_js_marshalled (https://optimizer.ml/_framework/wasm/dotnet.3.2.0-rc1.20222.2.js:1:171485)
    at do_icall (wasm-function[6049]:0x111976)
    at do_icall_wrapper (wasm-function[1896]:0x52c3e)
    at interp_exec_method (wasm-function[1120]:0x25a03)
    at interp_runtime_invoke (wasm-function[5654]:0xf944e)
    at mono_jit_runtime_invoke (wasm-function[5108]:0xdfbfa)
    at do_runtime_invoke (wasm-function[1410]:0x3db59)
    at mono_runtime_try_invoke (wasm-function[418]:0xcfd4)
```

https://github.com/jameschch/LeanParameterOptimization/commit/09d0323392bd1ff148b99969006916ea1bae9d1d

**Comment by jespersh** [comment]
I tried playing with this call before `NavigateTo`
```C#
await JSRuntime.InvokeVoidAsync("document.activeElement.blur");
```
but it only worked the first time I saved my form

**Comment by vstribrny** [comment]
Please notice there's the same issue also with `onfocusout` event. I am also facing this issue. The `onblur ` and `onfocusout` are basically unusable because you never know when user navigates back...

@SteveSandersonMS
Is there a chance to include a fix into some patch for 3.2?

**Comment by mkArtakMSFT** [comment]
@captainsafia the `contenteditable` issue may be related to this one. Please start from this one. If it is the same cause - great. If not, the other issue may need to be pushed out as it doesn't seem very critical.
